src/lsm/binary_search.zig

```diff
@@ -1,14 +1,19 @@
 const std = @import("std");
 const assert = std.debug.assert;
 const math = std.math;
+const config = @import("../config.zig");
 
 // TODO Add prefeching when @prefetch is available: https://github.com/ziglang/zig/issues/3600.
 //
 // TODO The Zig self hosted compiler will implement inlining itself before passing the IR to llvm,
 // which should eliminate the current poor codegen of key_from_value/compare_keys.
 
-/// Returns the index of the key either exactly equal to the target key or, if there is no exact
-/// match, the next greatest key.
+/// Returns either the index of the first value equal to `key`,
+/// or if there is no such value then the index where `key` would be inserted
+/// ie returns `i` such that both:
+/// * i == values.len or key_from_value(values[i]) >= key
+/// * i == 0 or key_value_from(values[i-1]) < key
+///
```

Comment:
I swapped the `or` order so that it aligned more, which makes it easier to spot similarities + differences at a glance.

```suggestion
/// Returns either the index of the first value equal to `key`,
/// or if there is no such value then the index where `key` would be inserted.
///
/// In other words, return `i` such that both:
/// * key_from_value(values[i])  >= key or i == values.len
/// * key_value_from(values[i-1]) < key or i == 0
///
```